lib/updates/market.ts

```diff
@@ -0,0 +1,167 @@
+import { getAddress } from 'ethers/lib/utils.js';
+import { BLOCKS_IN_A_DAY } from 'hooks/usePoolStats/usePoolStats';
+import { TargetUpdate } from 'hooks/useTarget';
+import { configs, SupportedToken } from 'lib/config';
+import { SECONDS_IN_A_YEAR } from 'lib/constants';
+import {
+  erc20Contract,
+  jsonRpcControllerContract,
+  makeProvider,
+} from 'lib/contracts';
+import { marks } from 'lib/controllers/charts/mark';
+import { targets } from 'lib/controllers/charts/target';
+import { formatDollars, formatPercent } from 'lib/numberFormat';
+import { percentChange } from 'lib/tokenPerformance';
+import {
+  TargetUpdatesByControllerDocument,
+  TargetUpdatesByControllerQuery,
+} from 'types/generated/graphql/inKindSubgraph';
+import {
+  PoolByIdByBlockDocument,
+  PoolByIdByBlockQuery,
+  PoolByIdDocument,
+  PoolByIdQuery,
+  SwapsByPoolDocument,
+  SwapsByPoolQuery,
+  Token,
+} from 'types/generated/graphql/uniswapSubgraph';
+import { createClient } from 'urql';
+
+export async function getTargetsInfo(token: SupportedToken) {
+  const provider = makeProvider(configs[token].jsonRpcProvider, token);
+  const currentBlock = await provider.getBlock('latest');
+
+  const newTarget = await jsonRpcControllerContract(
+    configs[token].controllerAddress,
+    token,
+  ).newTarget({ blockTag: currentBlock.number });
+
+  const underlyingDecimals = await erc20Contract(
+    configs[token].underlyingAddress,
+    token,
+  ).decimals();
```

Comment:
easier to get this from the subgraph and save RPC calls? 🤔

Reply:
you make a graphql below so could combine

Reply:
true, think its more readable to do this than entire graphQL call, we're on the server and tweeting every hour so performance less important?